transaction/types/src/lib.rs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Type-safe wrappers for integers used in our transactions, and other
+//! low-level types. This crate is intended to have a small footprint
+//! and be maximally portable.
+
+#![no_std]
+#![deny(missing_docs)]
+
+mod amount;
+mod token;
+
+pub use amount::Amount;
+pub use token::TokenId;
```

Comment:
I've seen this pattern where we have `workspace/types` starting to evolve organically - is this the pattern that we're more or less deciding on for types?

Reply:
yah i guess so

I think because of how rust and orphan rules works, it's often really good to have

1.) Very low-level types used often in APIs
2.) Traits

in their own crate(s) near the bottom of the totem pole, so that they can be maximally portable, and other maximally portable APIs can use them without bringing in any unwanted dependencies